packages/fetch-http-handler/src/fetch-http-handler.ts

```diff
@@ -130,4 +130,16 @@ export class FetchHttpHandler implements HttpHandler {
     }
     return Promise.race(raceOfPromises);
   }
+
+  updateHttpClientConfig(key: keyof FetchHttpHandlerConfig, value: FetchHttpHandlerConfig[typeof key]): void {
+    this.config = undefined;
```

Comment:
Is it safe to just "discard" `this.config`'s value?

Reply:
Not sure what you mean by "discard". `this.config` needs to set to `undefined` here because `handle` method call will populate `this.config` (with the updated config property) if it is undefined.

See https://github.com/awslabs/smithy-typescript/blob/main/packages/fetch-http-handler/src/fetch-http-handler.ts#L40-L42